mgmt_cli_test.py

```diff
@@ -506,6 +512,86 @@ def test_backup_rate_limit(self):
         self.verify_backup_success(mgr_cluster=mgr_cluster, backup_task=backup_task)
         self.log.info('finishing test_backup_rate_limit')
 
+    @staticmethod
+    def _get_all_snapshot_files_s3(cluster_id, bucket_name, region_name):
+        file_set = set()
+        try:
+            s3_client = boto3.client('s3', region_name=region_name)
+            paginator = s3_client.get_paginator('list_objects')
+            pages = paginator.paginate(Bucket=bucket_name, Prefix=f'backup/sst/cluster/{cluster_id}')
+            for page in pages:
+                content_list = page["Contents"]
+                file_set.update([item["Key"] for item in content_list])
+        except KeyError as err:
+            if err.args[0] == "Contents":
+                pass  # No snapshot files of the cluster exist, probably no backup ran before this function
+            else:
+                raise
+        return file_set
+
+    @staticmethod
+    def _get_all_snapshot_files_gce(cluster_id, bucket_name):
+        file_set = set()
+        gcp_credentials = KeyStore().get_gcp_credentials()
+        gce_driver = libcloud.storage.providers.get_driver(libcloud.storage.types.Provider.GOOGLE_STORAGE)
```

Comment:
libcloud should be replaced with google native APIs.
That is since it is a 3rd party limited indirect and a risky library for compatibility.
See how we addressed it in siren-tests repository, for example by using:
`import googleapiclient.discovery
from google.oauth2 import service_account`

Reply:
I tried to replace it, but it seems we use different forms of credentials (no client_email, etc.). For now, I'll stick with that.